scripts/uniqify-module-names.py

```diff
@@ -55,46 +54,144 @@ def generate_copy(c, sfx):
   bash(f"sed -i s/\"module {cur_name}\"/\"module {new_name}\"/ {new_file}")
   return new_file
 
-def dfs_update_modules(tree, common_fnames, visited, top_fnames):
+def dfs_update_modules(tree, common_fnames, visited, ext_dict):
   # List of direct submodules to update
   childs_to_update = list()
   for child in tree['instances']:
     # We don't have to change stuff that are under the dut
-    if (child['module_name'] == args.dut) or (child['module_name'] in visited):
+    if (child['module_name'] == args.dut):
       continue
-    if dfs_update_modules(child, common_fnames, visited, top_fnames):
+    if dfs_update_modules(child, common_fnames, visited, ext_dict):
       childs_to_update.append(child['module_name'])
-      if (child['module_name'] + ".sv") in common_fnames:
+      if (child['module_name']) in common_fnames:
         child['module_name'] = child['module_name'] + "_" + MODEL_SFX
 
   cur_module = tree['module_name']
-  cur_file = cur_module + ".sv"
   new_file = None
 
   # cur_file is in the common list, or is a ancestor of of them, generate a new file
-  if (cur_file in common_fnames) or len(childs_to_update) > 0:
-    new_file = generate_copy(cur_file, MODEL_SFX)
-    update_filelist(cur_file, os.path.basename(new_file))
-
-  for submodule_name in childs_to_update:
-    if (submodule_name + ".sv") in common_fnames:
-      bash(f"sed -i s/\"{submodule_name}\"/\"{submodule_name}_{MODEL_SFX}\"/ {new_file}")
+  if (cur_module in common_fnames) or len(childs_to_update) > 0:
+    new_file = 1
 
   visited.add(cur_module)
   return (new_file is not None)
 
+def bfs_update(tree, common_fnames, ext_dict, filelist):
+  q = [(tree['instance_name'], tree['module_name'], tree['instances'], None)]
+
+  updated_submodule = set()
+
+  while len(q) != 0:
+    front = q[0]
+    q.pop(0)
+    (inst, mod, child, parent) = front
+
+    try:
+      cur_file = mod + "." + ext_dict[mod]
+    except:
+      cur_file = mod + ".sv"
+
+    mod_updated = False
+
+    # if the module is common, make a copy & update its instance in its parent
+    if mod in common_fnames:
+      mod_updated = True
+      new_file = generate_copy(cur_file, MODEL_SFX)
+      filelist.append(new_file)
+      if parent is not None and ((parent, mod) not in updated_submodule):
+        print(mod, parent)
+        parent_file = os.path.join(args.gcpath, parent + "." + ext_dict[parent])
+        bash(f"sed -i s/\"{mod}\"/\"{mod}_{MODEL_SFX}\"/ {parent_file}")
```

Comment:
I think this needs to be more specific to target only the instance declaration. for example, this is what comes out when I grep for a uniquified module in TestHarness:

```
TestHarness.sv:  wire        _plusarg_reader_TestHarness_UNIQUIFIED_1_out;      // @[PlusArg.scala:80:11]
TestHarness.sv:  wire [31:0] _plusarg_reader_TestHarness_UNIQUIFIED_out;        // @[PlusArg.scala:80:11]
TestHarness.sv:    .custom_boot                   (_plusarg_reader_TestHarness_UNIQUIFIED_1_out),       // @[PlusArg.scala:80:11]
TestHarness.sv:    .enable          (_plusarg_reader_TestHarness_UNIQUIFIED_out[0]),    // @[Periphery.scala:227:18, PlusArg.scala:80:11]
TestHarness.sv:  plusarg_reader_TestHarness_UNIQUIFIED #(
TestHarness.sv:  ) plusarg_reader_TestHarness_UNIQUIFIED (      // @[PlusArg.scala:80:11]
TestHarness.sv:    .out (_plusarg_reader_TestHarness_UNIQUIFIED_out)
TestHarness.sv:  plusarg_reader_TestHarness_UNIQUIFIED #(
TestHarness.sv:  ) plusarg_reader_TestHarness_UNIQUIFIED_1 (    // @[PlusArg.scala:80:11]
TestHarness.sv:    .out (_plusarg_reader_TestHarness_UNIQUIFIED_1_out)
```

I think you can solve this by doing `sed -i s/\"{mod} \"/\"{mod}_{MODEL_SFX} \"/ {parent_file}` (note the extra space after the module name).